server.js

```diff
@@ -1,24 +1,68 @@
-var express = require("express");
-var mongoose = require("mongoose");
+//////////////
+//Importing//
+////////////
 
-require("./source/mongoose.connection");
-require("./source/mongoose.schemas");
+var http = require("http")
+var express = require("express")
+var socketio = require("socket.io")
 
-var app = express();
+var AssetStore = require("./source/asset.store.js")
+var YoutubeUtils = require("./source/youtube.process.js")
 
-app.use("/v2", require("./source/router.js"));
-app.use("/greet", require("./source/greet.router.js"));
+////////////
+//Routing//
+//////////
 
-app["all"]("*", function(request, response)
+application = express()
+application["use"]("/v2", require("./source/router.js"))
+application["use"]("/", require("./source/greet.router.js"))
+application["all"]("*", function(request, response)
 {
-    response.status(404).send("put error message here");
-});
+    response.status(404).send("put error message here")
+})
+
+////////////
+//Serving//
+//////////
 
-var port = process.env.PORT || 8080;
-var server = app.listen(port, function()
+port = process.env.PORT || 8080
+server = http.Server(application)
+server.listen(port, function()
 {
-    var host = server.address().address
-    var port = server.address().port
+    console.log("Cumulonimbus is serving at " + port)
+})
+
+//////////////
+//Streaming//
+////////////
 
-    console.log("Example app listening at http://%s:%s", host, port)
+io = socketio(server)
+io.on("connection", function(socket)
+{
+    AssetStore.forEachAsset(function(asset)
+    {
+        socket.emit("add asset", asset)
+    })
```

Comment:
Yeah! We use this to communicate to the web client frontend using Socket.IO, like this: https://github.com/viditor/chrono/blob/master/source/index.js